lib/app/dispatcher.rb

```diff
@@ -93,9 +105,36 @@ def start_polling_jobs!
         Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
       end
 
-      def send_heartbeat(connector_id, service_type)
-        Utility::Logger.info("Sending a heartbeat on [#{connector_id} - #{service_type}]...")
-        Core::Heartbeat.send(connector_id, service_type)
+      def start_heartbeat_task!
+        Thread.new do
+          Utility::Logger.info('Start heartbeat task...')
```

Comment:
I really don't like the code repetition in the `start_xxx_task` methods. I think it's a clear case for refactoring.

Reply:
I don't see it necessary to refactor here. the only part is duplicating is this
```
Utility::Logger.info('Start heartbeat task...')
heartbeat_scheduler.when_triggered do |connector_settings|
  ...
rescue StandardError => e
  Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
end
```
So it may be rafactored to 
```
def start_task(type)
  Utility::Logger.info("Start #{type} task...")
  send("#{type}_scheduler".to_sym).when_triggered do |connector_settings|
    yield connector_settings
  rescue StandardError => e
    Utility::ExceptionTracking.log_exception(e, "#{type} task failed due to unexpected error.")
  end
end
```
But I don't really think this is necessary.